add a `.clang-format` file and format the repo

AuroraPerego opened the pull request:
BEGINRELEASENOTES
- add a `.clang-format` file and format the repo

ENDRELEASENOTES

tmadlener reviewed (commented):
Thanks a lot for this. Since we have a [common `.clang-format`](https://github.com/key4hep/key4hep-dev-utils/blob/main/defaults/.clang-format) to ensure consistent formatting for all Key4hep repositories, I would prefer if we used that also here, unless there are compelling reasons not to do that.

AuroraPerego commented:
> Thanks a lot for this. Since we have a [common `.clang-format`](https://github.com/key4hep/key4hep-dev-utils/blob/main/defaults/.clang-format) to ensure consistent formatting for all Key4hep repositories, I would prefer if we used that also here, unless there are compelling reasons not to do that.

Oh, I did not know that :)
I just used one I was already using in another place, I can move to this one.

jmcarcell commented:
I updated the common `.clang-format`, reformatted the repository and added the commit to `.git-blame-ignore-revs`. Can you try to exctract the changes https://github.com/key4hep/k4Clue/pull/67 that are not formatting and rebase? I would not try to solve the conflicts, just simply try to apply the changes again (I think it's easier?).

AuroraPerego commented:
> I updated the common `.clang-format`, reformatted the repository and added the commit to `.git-blame-ignore-revs`. Can you try to exctract the changes #67 that are not formatting and rebase? I would not try to solve the conflicts, just simply try to apply the changes again (I think it's easier?).

done!
